### ISSUE TITLE
System.Threading.Tasks.Dataflow needs version 4.5.24.0 on NuGet.org

### ISSUE THREAD

**Issue by jaredpar** [dotnet/corefx#14869]
The lowest available release version of System.Threading.Tasks.Dataflow on NuGet.org is 4.5.25.0.  This creates a problem when building / testing Visual Studio components as it ships with version 4.5.24.0.  

Can we get an official package for 4.5.24.0 uploaded to NuGet.org?  This would simplify and give us greater confidence in our testing.

**Comment by tannergooding** [comment]
The package is available in a v4.5.24, but it had a different name at that time: https://www.nuget.org/packages/Microsoft.Tpl.Dataflow/4.5.24

This different name does break the transitive reference stuff in project.json/nuget v3.

Ideally, the Microsoft.Tpl.Dataflow packages would be repacked with the correct name and re-uploaded to nuget.

**Comment by jaredpar** [comment]
> Ideally, the Microsoft.Tpl.Dataflow packages would be repacked with the correct name and re-uploaded to nuget.

Agree.  As you pointed out this is necessary to make transitive dependencies sane.

**Comment by jaredpar** [comment]
This is starting to cause us a bit of pain in Roslyn.  Our build graph is now getting this reference from both package names.  Until there is a 4.5.24.0 there is no way to unify other than inserting a lot of suppressParent / include: none into the project.json files.  

The 4.5.24.0 is the most popular version.  Can we get this uploaded so we can standardize and fix the broken NuGet graphs?

**Comment by stephentoub** [comment]
cc: @ericstj, @weshaggard

**Comment by jaredpar** [comment]
Ping ... @ericstj, @weshaggard

**Comment by weshaggard** [comment]
I have no context on this package, beyond we shipped it under a new identity when we moved it to corefx. Perhaps @ericstj has more then I.

However I would like to better understand the reason you need a System.Threading.Task.Dataflow  4.5.24.0 package? I'm not even entirely sure we can ship one outside of manually hacking up the package and reshipping the same binaries. 

If you need to stick with the 4.5.24.0 version what issues are caused by continuing to reference Microsoft.Tpl.Dataflow 4.5.24 like you did here https://github.com/jaredpar/roslyn/commit/2c7fd95a93c3476caa515794aa131992cd9b9931?

**Comment by jaredpar** [comment]
@weshaggard the problem is that it creates conflicts in our graph when a component depends on a newer version of Dataflow.  The only solution we have is to liberally spray "suppressParent" entries into our project.json files to avoid a slew of build and restore warnings.  

There is a single DLL here that we, as Microsoft, have chosen to release under two different identities.  This makes it very difficult for consumers to work with the package, especially when we need to have a staged move to a new version.  

>I'm not even entirely sure we can ship one outside of manually hacking up the package and reshipping the same binaries.

What's the problem with this approach?  Just download the Tpl version, create a NuGet based on the DataFlow NuSpec and upload.

**Comment by weshaggard** [comment]
>@weshaggard the problem is that it creates conflicts in our graph when a component depends on a newer version of Dataflow. The only solution we have is to liberally spray "suppressParent" entries into our project.json files to avoid a slew of build and restore warnings.

Even if we upload a the 4.5.24.0 version of System.Threading.Task.Dataflow package won't this still be true? Or will you then force everyone to switch their package reference to System.Threading.Task.Dataflow?

**Comment by jaredpar** [comment]
@weshaggard 

> Or will you then force everyone to switch their package reference to System.Threading.Task.Dataflow?

That is the idea.  By having a version of System.Threading.Task.Dataflow uploaded at the same version we can stage the migration to this package.

Owners of packages that depend on MS.Tpl.Dataflow can release a new version that just changes the package dependency from MS.Tpl.Dataflow to System.Threading.Tasks.Dataflow.  As the assemblies are identical in both packages down stream consumers can make the switch without having to worry about assembly version merges.  For example I don't have to worry about ramifications to setup authoring / central version management in Visual Studio.  

After that the package owners can begin migrating to newer versions of System.Threading.Tasks.Dataflow without creating chaos in all the consumers NuGet graphs.

**Comment by weshaggard** [comment]
Sorry for the delay I just got back to this. I've repackaged the 4.5.24 Microsoft.tpl.dataflow package as 4.5.24 System.Threading.Tasks.DataFlow however beyond basic file content comparison I don't have a great way to verify this package. Before I push this package to nuget.org (which cannot easily be undone) I've uploaded the package to our dotnet.myget.org feed. 

Could any of you guys that are interested in this try using the package from https://dotnet.myget.org/F/dotnet-core/api/v3/index.json (https://dotnet.myget.org/F/dotnet-core/api/v2/package/System.Threading.Tasks.Dataflow/4.5.24)? Once I get some further confirmation I will push the package to nuget.org.
